### PR TITLE
#9009 Renew/Upgrade payment: remove previous incomplete payment after a new payment created

### PR DESCRIPTION
**Pull request by eve-git**
Delete the all the incomplete payments after a new payment records created for Renew/Upgrade payments.

*Issue #:* /bcgov/entity#9009

*Description of changes:*
After a new renew or upgrade payment record created, delete all the previous incomplete payment records.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namerequest license (Apache 2.0).

### CODE REVIEW

**Comment by severinbeauvais** [comment]
@eve-git I assigned you so that the CI checks will run.

**Review by severinbeauvais — COMMENTED** [review]
Looks OK to me.

1. Please update the app version (in package.json, then run `npm i`).
2. I'd like to see some Test Notes in the ticket, and some evidence that this has been tested and works as intended.

File: src/mixins/payment-mixin.ts [hunk 1]
```diff
@@ -339,6 +339,20 @@ export class PaymentMixin extends Mixins(ActionMixin) {
...
       await this.setPaymentRequest(data)
 
+      // delete previous incomplete payments after the new payment record created
+      const allPayments = await this.getNameRequestPayments(nrId, {})
+      if (allPayments) {
+        for (let i = 0; i < allPayments.length; i++) {
+          if (
+            allPayments[i].payment.id !== payment.id &&
+            allPayments[i].payment.payment_action === action &&
+            allPayments[i].sbcPayment.statusCode === SbcPaymentStatus.CREATED
```
Comment (cameron-freshworks) [inline src/mixins/payment-mixin.ts:349]: Are these caught gracefully in the `try/catch` in the event one of these values is not available?
For instance, if `allPayments[i].payment.payment_action` is missing `payment`. 

If not, optional chaining might a solution. Up to you!
eg `allPayments[i].payment?.payment_action`

**Review by cameron-freshworks — COMMENTED** [review]
Code looks good, just a question/comment

**Review by cameron-freshworks — APPROVED** [review]

**Comment by eve-git** [comment]
Since Renew and Upgrade share one piece of code to create a new payment, I am only list the steps to reproduce the error for Upgrade:
1. request a name request without priority
2. upgrade priority
> a. click Upgrade Priority in nr search result page
> b. click Continue to Payment in payment confirmation page
> c. leave the credit payment page by close the browser, or click "back" buttons in browser or input a new url in the browser
(The error won't show up by cancel the payment (cancel in both credit card input page and payment cancelled page)
3. search the nr in #1, the result page shows Request Status as Payment Incomplete.
4. try upgrade priority again, and finish all the payment steps
5. repeat #3 and go to NR search result page. Even though the nr has been upgraded successfully (you can see two receipt), the Request Status is still Payment Incomplete.

Test:
follow the steps 1-4 above. When back to search result page, Request Status is Staff Review instead of Payment Incomplete.

**Review by severinbeauvais — APPROVED** [review]
LGTM. Thanks for the updates!

**Pull request merged**